platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -964,13 +964,13 @@ void OfflineDatabase::putRegionResources(int64_t regionID,
 }
 
 uint64_t OfflineDatabase::putRegionResourceInternal(int64_t regionID, const Resource& resource, const Response& response) {
-    if (exceedsOfflineMapboxTileCountLimit(resource)) {
-        throw MapboxTileLimitExceededException();
-    }
-
     uint64_t size = putInternal(resource, response, false).second;
     bool previouslyUnused = markUsed(regionID, resource);
 
+    if (previouslyUnused && exceedsOfflineMapboxTileCountLimit(resource)) {
```

Comment:
The logic has changed from initial version. Now, we insert first and then check if tiles count is over the limit. Should the count limit be checked before `putInternal / markUsed`?

Reply:
That was another bug. If we check before, we cannot know if the tile we are trying to insert is overlapping with another region, and in this case, not counting as a tile in the total tile count.

The insertion is inside a SQL transaction. If we throw here it will rollback and the tile will not get inserted.

Reply:
:+1: